packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -119,27 +282,76 @@ export const OAuthLoginPage = () => {
     }
     return null
   }, [redirect_uri])
-  const account = useSelector(getAccountUser)
-  const isLoggedIn = Boolean(account)
-  const [userEmail, setUserEmail] = useState<string | null>(null)
-  useEffect(() => {
-    const getAndSetEmail = async () => {
-      let email: string
+
+  const isRedirectValid = useMemo(() => {
+    return getIsRedirectValid({ parsedRedirectUri, redirectUri: redirect_uri })
+  }, [parsedRedirectUri, redirect_uri])
+
+  const parsedOrigin = useMemo(() => {
+    if (originParam && typeof originParam === 'string') {
       try {
-        email = await audiusBackendInstance.getUserEmail()
+        return new URL(originParam)
       } catch {
-        setUserEmail(null)
-        history.push(ERROR_PAGE)
-        return
+        return null
       }
-      setUserEmail(email)
     }
-    if (isLoggedIn) {
-      getAndSetEmail()
-    } else {
-      setUserEmail(null)
+    return null
+  }, [originParam])
+
+  let initError: string | null = null
```

Comment:
def a minor refactor  thing, would would be just making a "getError" function or something rather?